src/components/YoutubePlayer.jsx

```diff
@@ -0,0 +1,11 @@
+import ReactPlayer from 'react-player'
+
+const YoutubePlayer = ({ videoKey }) => (<ReactPlayer 
```

Comment:
The video should open in a modal.
Include a close button.
Pass the onClose prop to ensure the modal can be closed when clicking outside the video (including necessary styles).
Prevent the propagation of the onClick event when clicking on the video to avoid inadvertently closing the modal.